src/reaktoro_pse/tutorials/softening_acid_ro_example.ipynb

```diff
@@ -0,0 +1,1324 @@
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = ConcreteModel()\n",
+    "# create IDAES flowsheet\n",
+    "m.fs = FlowsheetBlock(dynamic=False)\n",
+    "# can use NaCl or Sea water prop pack"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Simple feed block "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get ions\n",
+    "ions = list(sea_water_composition.keys())\n",
+    "#\n",
+    "m.fs.feed = Block()\n",
+    "m.fs.feed.species_concentrations = Var(\n",
+    "    ions, initialize=1, bounds=(0, None), units=pyunits.mg / pyunits.L\n",
+    ")\n",
+    "# ReaktoroBlock needs mass flows of all the species\n",
+    "ions.append(\"H2O\")\n",
+    "m.fs.feed.species_mass_flow = Var(\n",
+    "    ions, initialize=1, bounds=(0, None), units=pyunits.kg / pyunits.s\n",
+    ")\n",
+    "\n",
+    "# ion mw\n",
+    "m.fs.feed.specie_mw = Param(ions, units=pyunits.kg / pyunits.mol)\n",
+    "m.fs.feed.specie_mw[\"Ca\"] = 40.078 / 1000\n",
+    "m.fs.feed.specie_mw[\"HCO3\"] = 61.0168 / 1000\n",
+    "m.fs.feed.specie_mw[\"Na\"] = 22.989769 / 1000\n",
+    "m.fs.feed.specie_mw[\"H2O\"] = 18.01528 / 1000\n",
+    "m.fs.feed.specie_mw[\"Mg\"] = 24.305 / 1000\n",
+    "m.fs.feed.specie_mw[\"Cl\"] = 35.453 / 1000\n",
+    "m.fs.feed.specie_mw[\"SO4\"] = 96.06 / 1000\n",
+    "# Create pH variable for feed\n",
+    "m.fs.feed.pH = Var(initialize=sea_water_ph)\n",
+    "m.fs.feed.pH.fix()\n",
+    "\n",
+    "# Create temperature variable for feed\n",
+    "m.fs.feed.temperature = Var(initialize=293, units=pyunits.K)\n",
+    "m.fs.feed.temperature.fix()\n",
+    "\n",
+    "# Create pressure variable for feed\n",
+    "m.fs.feed.pressure = Var(initialize=1e5, units=pyunits.Pa)\n",
+    "m.fs.feed.pressure.fix()\n",
+    "\n",
+    "# To get true flow mass, the density is needed for the solution, we can get that from reaktoro\n",
+    "m.fs.feed.density = Var(\n",
+    "    initialize=1053, units=pyunits.kg / pyunits.m**3\n",
+    ") \n",
+    "m.fs.feed.density.fix()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Write constraints to convert concentration to mass flows\n",
+    "@m.fs.feed.Constraint(list(m.fs.feed.species_mass_flow.keys()))\n",
+    "def eq_feed_species_mass_flow(fs, ion):\n",
+    "    if ion == \"H2O\":\n",
+    "        return Constraint.Skip\n",
+    "    else:\n",
+    "        \"\"\"calculate mass flow based on density\"\"\"\n",
+    "        return m.fs.feed.species_mass_flow[ion] == pyunits.convert(\n",
+    "            m.fs.feed.species_concentrations[ion]\n",
+    "            * m.fs.feed.species_mass_flow[\"H2O\"]\n",
+    "            / m.fs.feed.density,\n",
+    "            to_units=pyunits.kg / pyunits.s,\n",
+    "        )"
+   ]
+  },
```

Comment:
Cram these all into an aggregated cell for Step 1, setting up the feed. Under the "Step 1" markdown, you can outline the high-level operations that are occuring in this cell:
e.g., 

- build pyomo model
- specify feed conditions (can list feed conditions)
-  construct mass flow equations to convert concentrations